addons/mail/static/src/components/autocomplete_input/autocomplete_input.js

```diff
@@ -12,25 +14,23 @@ export class AutocompleteInput extends Component {
      */
     setup() {
         super.setup();
+        useComponentToModel({ fieldName: 'component', modelName: 'AutocompleteInputView' });
+        useUpdateToModel({ methodName: 'onComponentUpdate', modelName: 'AutocompleteInputView' });
         onMounted(() => this._mounted());
         onWillUnmount(() => this._willUnmount());
     }
 
     _mounted() {
-        if (this.props.isFocusOnMount) {
-            this.root.el.focus();
-        }
-
         let args = {
             autoFocus: true,
-            select: (ev, ui) => this._onAutocompleteSelect(ev, ui),
-            source: (req, res) => this._onAutocompleteSource(req, res),
-            focus: ev => this._onAutocompleteFocus(ev),
-            html: this.props.isHtml || false,
+            select: this.autocompleteInputView.onSelect,
+            source: this.autocompleteInputView.onSource,
+            focus: ev => ev.preventDefault(),
```

Comment:
how comes this `focus` param exists (and was used before), if we are now able to just use `t-on-focusin`?

Reply:
> how comes this `focus` param exists (and was used before), if we are now able to just use `t-on-focusin`?

if you look carefully at red diff, props `focus` was actually never used. I think it was provided just in case we had a need for it, but in the end we don't. In fact we don't want the default behaviour of `ui-autocomplete` focus on item, which is to set content of input as the focused item.
https://api.jqueryui.com/autocomplete/#event-focus

Reply:
maybe it's worth adding it as comment on the line to explain why